executor/aggregate.go

```diff
@@ -41,24 +42,96 @@ type HashAggExec struct {
 	aggCtxsMap    aggCtxsMapper
 	groupMap      *mvmap.MVMap
 	groupIterator *mvmap.Iterator
+	inputRow      chunk.Row
 	GroupByItems  []expression.Expression
 }
 
 // Close implements the Executor Close interface.
 func (e *HashAggExec) Close() error {
+	if err := e.baseExecutor.Close(); err != nil {
+		return errors.Trace(err)
+	}
 	e.groupMap = nil
 	e.groupIterator = nil
 	e.aggCtxsMap = nil
-	return errors.Trace(e.children[0].Close())
+	return nil
 }
 
 // Open implements the Executor Open interface.
 func (e *HashAggExec) Open(goCtx goctx.Context) error {
+	if err := e.baseExecutor.Open(goCtx); err != nil {
+		return errors.Trace(err)
+	}
 	e.executed = false
 	e.groupMap = mvmap.NewMVMap()
 	e.groupIterator = e.groupMap.NewIterator()
 	e.aggCtxsMap = make(aggCtxsMapper, 0)
-	return errors.Trace(e.children[0].Open(goCtx))
+	e.inputRow = e.childrenResults[0].End()
```

Comment:
maybe it's better for understanding to set `e.inputRow = e.childrenResults[0].Begin()` ?

Reply:
ok